### PR TITLE
Add table aws_ec2_client_vpn_endpoint closes #1717

### LINKED ISSUE #1717: Add table aws_ec2_client_vpn_endpoint
**References**

https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateClientVpnEndpoint.html
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_client_vpn_endpoint

### PR DESCRIPTION
**Pull request by misraved**
# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
select * from aws_ec2_client_vpn_endpoint
[
 {
  "_ctx": {
   "connection_name": "default"
  },
  "account_id": "112233445566",
  "authentication_options": [
   {
    "ActiveDirectory": null,
    "FederatedAuthentication": null,
    "MutualAuthentication": {
     "ClientRootCertificateChain": "arn:aws:acm:us-east-1:112233445566:certificate/68ed52d9-ae3e-4543-a2d5-112233445566"
    },
    "Type": "certificate-authentication"
   }
  ],
  "client_cidr_block": "10.0.0.0/10",
  "client_connect_options": {
   "Enabled": false,
   "LambdaFunctionArn": null,
   "Status": {
    "Code": "applying",
    "Message": null
   }
  },
  "client_log_options": null,
  "client_login_banner_options": {
   "BannerText": null,
   "Enabled": false
  },
  "client_vpn_endpoint_id": "cvpn-endpoint-04b9716701d11b1d3",
  "creation_time": "2023-05-15T13:47:12+05:30",
  "deletion_time": null,
  "description": "delete if found free",
  "dns_name": "*.cvpn-endpoint-04b9716701d11b1d3.prod.clientvpn.us-east-1.amazonaws.com",
  "dns_servers": null,
  "partition": "aws",
  "region": "us-east-1",
  "security_group_ids": [],
  "self_service_portal_url": null,
  "server_certificate_arn": "arn:aws:acm:us-east-1:112233445566:certificate/68ed52d9-ae3e-4543-a2d5-112233445566",
  "session_timeout_hours": 24,
  "split_tunnel": false,
  "status": {
   "Code": "pending-associate",
   "Message": null
  },
  "tags": {
   "Name": "test-01"
  },
  "tags_src": [
   {
    "Key": "Name",
    "Value": "test-01"
   }
  ],
  "title": "test-01",
  "transfer_protocol": null,
  "vpc_id": "",
  "vpn_port": 443,
  "vpn_protocol": "openvpn"
 }
]
```
</details>

### CODE REVIEW

File: aws/table_aws_ec2_client_vpn_endpoint.go [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+		ClientVpnEndpointIds: []string{clientVpnEndpointId},
+	}
+
+	op, err := svc.DescribeClientVpnEndpoints(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_ec2_client_vpn_endpoint.getEC2ClientVPNEndpoint", "api_error", err)
+		return nil, err
+	}
+
+	return op.ClientVpnEndpoints, nil
```
Comment (ParthaI) [inline aws/table_aws_ec2_client_vpn_endpoint.go:253]:
```suggestion
	return op.ClientVpnEndpoints[0], nil
```

**Review by ParthaI — CHANGES_REQUESTED** [review]
@misraved please take a look at the review comments. Thanks!

File: aws/table_aws_ec2_client_vpn_endpoint.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	// Limiting the results
+	maxLimit := int32(1000)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 5 {
+				maxLimit = 5
+			} else {
+				maxLimit = limit
+			}
```
[the remark this answers is not in the document]
Comment (misraved) [reply aws/table_aws_ec2_client_vpn_endpoint.go:187]: @ParthaI the API docs mention that the minimum limit should be 5

File: aws/table_aws_ec2_client_vpn_endpoint.go [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+//// TABLE DEFINITION
+
+func tableAwsEC2ClientVPNEndpoint(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_ec2_client_vpn_endpoint",
+		Description: "AWS EC2 Client VPN Endpoint",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("client_vpn_endpoint_id"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ValidationError", "InvalidQueryParameter", "InvalidParameterValue", "InvalidClientVpnEndpointId.NotFound"}),
```
Comment (ParthaI) [inline aws/table_aws_ec2_client_vpn_endpoint.go:26]: Should we ignore `ValidationError`, `InvalidQueryParameter`, `InvalidParameterValue `?

**Review by ParthaI — COMMENTED** [review]
@misraved Please take a look at the review comments, Thanks!
Could you please re-order the columns?

- First 3-4 column would be the important column for the table like (name, id, arn, status...).
- Re-order the Non-JSON column alphabetically.
- Re-order the JSON column alphabetically.

**Review by ParthaI — APPROVED** [review]
LGTM

**Pull request merged**